src/cargo/core/resolver/mod.rs

```diff
@@ -802,6 +820,18 @@ struct BacktrackFrame {
     conflicting_activations: HashMap<PackageId, ConflictReason>,
 }
 
+/// A helper "iterator" used to extract candidates within a current `Context` of
+/// a dependency graph.
+///
+/// This struct doesn't literally implement the `Iterator` trait (requires a few
+/// more inputs) but in general acts like one. Each `RemainingCandidates` is
+/// created with a list of candidates to choose from. When attempting to iterate
+/// over the list of candidates only *valid* candidates are returned. Validity
+/// is defined within a `Context`.
+///
+/// Candidates passed to `new` may not be returned from `next` as they could be
+/// filtered out, and if iteration stops a map of all failed candidates will be
```

Comment:
Nit: I think it returns a map of all conflicting already activated `PackageId` not a map of the filtered (not yet activated) candidates.

Reply:
Aha yes indeed!